qa/workunits/rbd/rbd_mirror_group.sh

```diff
@@ -0,0 +1,267 @@
+if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
+  wait_for_group_status_in_pool_dir ${CLUSTER2} ${POOL} ${group} 'down+unknown'
+fi
+
+testlog "TEST: add image to group and test replay"
+image=test-image
+create_image ${CLUSTER2} ${POOL} ${image}
+group_add_image ${CLUSTER2} ${POOL} ${group} ${POOL} ${image}
+wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group} 1
+wait_for_image_present ${CLUSTER1} ${POOL} ${image} 'present'
+write_image ${CLUSTER2} ${POOL} ${image} 100
+wait_for_group_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${group}
+wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group} 'up+replaying'
+compare_images ${POOL} ${image}
+
+testlog "TEST: test replay with remove image and later add the same"
+group_remove_image ${CLUSTER2} ${POOL} ${group} ${POOL} ${image}
+wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group} 0
+group_add_image ${CLUSTER2} ${POOL} ${group} ${POOL} ${image}
+wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group} 1
+
+testlog "TEST: stop mirror, create group, start mirror and test replay"
+stop_mirrors ${CLUSTER1}
+group1=test-group1
+create_group_and_enable_mirror ${CLUSTER2} ${POOL} ${group1}
+image1=test-image1
+create_image ${CLUSTER2} ${POOL} ${image1}
+group_add_image ${CLUSTER2} ${POOL} ${group1} ${POOL} ${image1}
+start_mirrors ${CLUSTER1}
+wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group1} 1
+wait_for_group_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${group1}
+wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group1} 'up+replaying'
+if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
+  wait_for_group_status_in_pool_dir ${CLUSTER2} ${POOL} ${group1} 'down+unknown'
+fi
+compare_images ${POOL} ${image1}
+
+testlog "TEST: test the first group is replaying after restart"
+write_image ${CLUSTER2} ${POOL} ${image} 100
+wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group} 1
+wait_for_group_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${group}
+wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group} 'up+replaying'
+compare_images ${POOL} ${image}
+
+if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
+  testlog "TEST: stop/start/restart group via admin socket"
+
+  admin_daemons ${CLUSTER1} rbd mirror group stop ${POOL}/${group1}
+  wait_for_group_replay_stopped ${CLUSTER1} ${POOL} ${group1} 1
+  wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group1} 'up+stopped'
+
+  admin_daemons ${CLUSTER1} rbd mirror group start ${POOL}/${group1}
+  wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group1} 1
+  wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group1} 'up+replaying'
+
+  admin_daemons ${CLUSTER1} rbd mirror group restart ${POOL}/${group1}
+  wait_for_group_replay_started ${CLUSTER1} ${POOL} ${group1} 1
+  wait_for_group_status_in_pool_dir ${CLUSTER1} ${POOL} ${group1} 'up+replaying'
+
+  flush ${CLUSTER1}
+  admin_daemons ${CLUSTER1} rbd mirror group status ${POOL}/${group1}
+fi
+
+testlog "TEST: test group rename"
```

Comment:
should we maintain same `group` names for both sites in mirroring?

Reply:
The user only needs to create the group on the primary cluster. The mirroring daemon will take care of creating a group with the same name on the secondary cluster (like is currently done for image mirroring).